Hello @kshitij10496! Thanks for submitting the PR.

 - In the file `stackdidact/cli.py`, following                     are the PEP8 issues :
```
stackdidact/cli.py:6:1: E302 expected 2 blank lines, found 1
stackdidact/cli.py:13:1: E305 expected 2 blank lines after class or function definition, found 1
```

 - In the file `stackdidact/generator.py`, following                     are the PEP8 issues :
```
stackdidact/generator.py:23:21: E261 at least two spaces before inline comment
stackdidact/generator.py:25:1: E302 expected 2 blank lines, found 1
stackdidact/generator.py:29:1: E302 expected 2 blank lines, found 1
stackdidact/generator.py:35:1: E302 expected 2 blank lines, found 1
stackdidact/generator.py:37:27: W291 trailing whitespace
stackdidact/generator.py:48:1: W293 blank line contains whitespace
stackdidact/generator.py:49:1: E302 expected 2 blank lines, found 1
stackdidact/generator.py:55:1: W293 blank line contains whitespace
stackdidact/generator.py:58:1: E302 expected 2 blank lines, found 1
stackdidact/generator.py:62:8: E713 test for membership should be 'not in'
stackdidact/generator.py:71:37: W291 trailing whitespace
stackdidact/generator.py:73:1: W293 blank line contains whitespace
```

 - In the file `stackdidact/question.py`, following                     are the PEP8 issues :
```
stackdidact/question.py:7:1: E302 expected 2 blank lines, found 1
stackdidact/question.py:27:19: E203 whitespace before ':'
```

--- a/stackdidact/cli.py
+++ b/stackdidact/cli.py
@@ -3,12 +3,14 @@ import click
 from generator import generate_question
 from settings import DATA
 
+
 @click.command()
 @click.option('--tag', default='python', help='Search tag')
 def main_cli(tag):
     DATA['tagged'] = tag
     question = generate_question(DATA)
     click.echo(question)
+
 
 if __name__ == "__main__":
     main_cli()

--- a/stackdidact/generator.py
+++ b/stackdidact/generator.py
@@ -20,11 +20,13 @@ except FileNotFoundError:
         }
         json.dump(id_data, fp)
 
-ids = id_data['yes'] # store the ids of the questions already displayed
+ids = id_data['yes']  # store the ids of the questions already displayed
+
 
 def request_site(data):
     response = requests.get(URL, data=data)
     return response.json()
+
 
 def format(question):
     print(Fore.CYAN + Style.BRIGHT + "Q: " + Fore.RED + Style.BRIGHT + question.title)
@@ -32,9 +34,10 @@ def format(question):
     print(Fore.MAGENTA + "OP: " + question.owner_name)
     print(Fore.GREEN + "Read the discussion :" + question.link)
 
+
 def generate_question(data):
     """
-    data : list of strings 
+    data : list of strings
     """
     temp_ids = get_ids(data)
     question_id = logic(temp_ids, data)
@@ -45,21 +48,23 @@ def generate_question(data):
 
     question = Question.from_id(question_id)
     return question
- 
+
+
 def get_ids(data):
     temp_ids = []
     response = request_site(data)
     question_set = response["items"]
     for question in question_set:
         temp_ids.append(question["question_id"])
-    
+
     return temp_ids
+
 
 def logic(temp_ids, data):
     """ Generate random question_id from list of ids.
     """
     question_id = random.choice(temp_ids)
-    if not question_id in ids:
+    if question_id not in ids:
         ids.append(question_id)
         return question_id
 
@@ -68,9 +73,9 @@ def logic(temp_ids, data):
         if len(ids) >= data['page'] * data['pagesize']:
             data['page'] += 1
 
-        new_temp_ids = get_ids(data) 
+        new_temp_ids = get_ids(data)
         return logic(new_temp_ids, data, URL)
-    
+
     else:
         new_temp_ids = temp_ids.remove(question_id)
         return logic(new_temp_ids, data)

--- a/stackdidact/question.py
+++ b/stackdidact/question.py
@@ -4,6 +4,7 @@ import requests
 
 from settings import URL, SITE
 
+
 class Question:
 
     def __init__(self, tags, owner_name, question_id, title, link):
@@ -24,7 +25,7 @@ class Question:
     def from_id(cls, question_id):
         main_url = URL + '/' + str(question_id)
         data = {
-            'site' : SITE,
+            'site': SITE,
             }
         response = requests.get(main_url, data=data).json()
         question_body = response['items'][0]

--- a/stackdidact/settings.py
+++ b/stackdidact/settings.py
@@ -5,11 +5,11 @@ SITE = 'stackoverflow'
 TAG = 'python'
 DATA = {
         'page': 1,
-        'pagesize': 30, # mentioning the default value for completeness
-        'order' : 'desc',
+        'pagesize': 30,  # mentioning the default value for completeness
+        'order': 'desc',
         'sort': 'votes',
-        'tagged': TAG, # Searches for questions using this tag
-        'site': SITE # search website in the Stack Exchange family
+        'tagged': TAG,  # Searches for questions using this tag
+        'site': SITE  # search website in the Stack Exchange family
     }
 PATH = os.path.dirname(os.path.abspath(__file__))
 BASE_DIR = os.path.split(PATH)[0]

